plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestSpatialJoinPlanning.java

```diff
@@ -301,30 +307,26 @@ public void testDistanceQuery()
                         "WHERE ST_Distance(ST_Point(a.lng, a.lat), ST_Point(b.lng, b.lat)) <= 3.1",
                 anyTree(
                         spatialJoin("st_distance(st_point_a, st_point_b) <= radius",
-                                project(ImmutableMap.of("st_point_a", expression("ST_Point(cast(a_lng as double), cast(a_lng as double))"), "a_name", expression("'x'")),
-                                        project(
-                                                ImmutableMap.of("a_lng", expression("CAST(DECIMAL '2.1' AS decimal(2, 1))")),
-                                                singleRow())),
+                                project(
+                                        ImmutableMap.of("st_point_a", expression(point21x21Literal), "a_name", expression("'x'")),
```

Comment:
Here & below. Previously we were projecting `CAST(DECIMAL '2.1' AS decimal(2, 1))` column to then construct `ST_Point(x,x)` from it. Now this got recognized as effectively literal and inlined. Thus `ST_Point()` function call is replaced with literal (`point21x21Literal`), and subsequent Project nodes are merged into one.